src/sentry/utils/distutils/commands/build_assets.py

```diff
@@ -127,6 +127,10 @@ def _build_static(self):
         env = dict(os.environ)
         env["SENTRY_STATIC_DIST_PATH"] = self.sentry_static_dist_path
         env["NODE_ENV"] = "production"
+        # TODO: Our JS builds should not require 4GB heap space
+        # XXX(byk): This would override any existing `NODE_OPTIONS` in the environment
+        #           which can break certain assumptions and hard to debug
+        env["NODE_OPTIONS"] = "--max-old-space-size=4096"
```

Comment:
Why not do like:

```python
env["NODE_OPTIONS"] = env.get("NODE_OPTIONS", "") + " --max-old-space-size=4096"
```

Reply:
Because then I'd need to add a left trim to be precise and this would clash with a potentially existing `--max-old-space-size=XXX` value. I figured we can afford to be strict here, even we _should_ be strict.

WDYT?

Reply:
Would an extra space make it not work? That seems silly.

```python
env["NODE_OPTIONS"] = " ".join([env.get("NODE_OPTIONS", ""), "--max-old-space-size=4096"])
```

:)

Reply:
Ehh, sure will do that :D

Reply:
Nevermind, you don't wanna duplicate. Yeah, I dunno, nor do I care that much. I don't think there's any other practical case where any of this would matter. We can just clobber it.

Reply:
Yeah doesn’t seem like a likely scenario where someone would be specifying other node options here